### PR TITLE
✨ NEW: Port `admon` plugin

### PR DESCRIPTION
**Pull request by KyleKing**
Fixes: https://github.com/executablebooks/mdformat/issues/309

I had two hours this morning and tried porting the plugin! It passes all tests locally and seems to work, but I'll take another look through tomorrow to make sure it is merge-ready. I'm opening a PR for now, since there seemed to be a couple of people interested, and I would benefit from any feedback.

### CODE REVIEW

File: mdit_py_plugins/admon/LICENSE [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2015 Vitaly Puzrin, Alex Kocharin.
+Copyright (c) 2018 jebbs
+Copyright (c) 2021- commenthol
+
+Permission is hereby granted, free of charge, to any person
```
Comment (KyleKing) [inline mdit_py_plugins/admon/LICENSE:5]: TBD: What license goes here?

File: mdit_py_plugins/admon/index.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    return tag.lower(), title
+
+
+def validate(params: str) -> bool:
+    tag = params.strip().split(" ", 1)[-1] or ""
+    if tag.lower() in ADMONITION_TAGS:
+        return True
+    # FIXME: Should this return False? Or better way to log warnings?
+    print(f"Warning: admonition tag {tag} is not one of: {ADMONITION_TAGS}")
+    return bool(tag)
```
Comment (KyleKing) [inline mdit_py_plugins/admon/index.py:60]: TBD: How strict should the validation be? The original JS plugin didn't check against the list of tags

File: mdit_py_plugins/admon/index.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    return True
+
+
+def admon_plugin(md: MarkdownIt) -> None:
+    """Plugin ported from:
+
+    `markdown-it-admon <https://github.com/commenthol/markdown-it-admon/blob/9820ba89415c464a3cc18a780f222a0ceb3e18bd/index.js>`.
+
+    """
+    md.block.ruler.before(
```
Comment (KyleKing) [inline mdit_py_plugins/admon/index.py:185]: TBD: the original plugin had options, and I'll need to follow up to see how to port those. For reference:

https://github.com/commenthol/markdown-it-admon/blob/9820ba89415c464a3cc18a780f222a0ceb3e18bd/index.js#L164-L170

**Comment by KyleKing** [comment]
I think this is now ready for review. Let me know if there is anything that needs to be modified, and I would be happy to make those changes, or feel free to go ahead and make changes directly!

Based [on this recommendation](https://github.com/executablebooks/mdformat/issues/309#issuecomment-1046327823), this PR ports this JS plugin to Python: https://github.com/commenthol/markdown-it-admon/blob/9820ba89415c464a3cc18a780f222a0ceb3e18bd/index.js

**Review by hukkin — CHANGES_REQUESTED** [review]
Thanks! This seems good to me assuming that tests pass.

Would @chrisjsewell be able to approve CI and review too?

File: mdit_py_plugins/admon/port.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+- package: markdown-it-admon
+  commit: 9820ba89415c464a3cc18a780f222a0ceb3e18bd
+  date: Nov 26, 2022
```
[the remark this answers is not in the document]
Comment (KyleKing) [reply mdit_py_plugins/admon/port.yaml:3]: Makes sense to me!

```suggestion
  date: Jul 3, 2021
```

**Comment by hukkin** [comment]
Regarding broken CI checks:
- both failures unrelated to this PR
- pre-commit check should be fixed by https://github.com/executablebooks/mdit-py-plugins/pull/54

File: mdit_py_plugins/admon/index.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+#     "fail",
+#     "missing",
+#     "danger",
+#     "error",
+#     "bug",
+#     "example",
+#     "snippet",
+#     "quote",
+#     "cite",
+# ]
```
Comment (chrisjsewell) [inline mdit_py_plugins/admon/index.py:38]: if this is not used, then I would just delete it

File: mdit_py_plugins/admon/index.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    return bool(tag)
+
+
+MIN_MARKERS = 3
+MARKER_STR = "!"
+MARKER_CHAR = ord(MARKER_STR)
+MARKER_LEN = len(MARKER_STR)
+
+
+def admonition(state: StateBlock, startLine: int, endLine: int, silent: bool):
```
Comment (chrisjsewell) [inline mdit_py_plugins/admon/index.py:67]:
```suggestion
def admonition(state: StateBlock, startLine: int, endLine: int, silent: bool) -> bool:
```

**Review by chrisjsewell — CHANGES_REQUESTED** [review]

**Comment by KyleKing** [comment]
Thanks for the feedback! I think I made all the requested changes. Let me know if there is anything else.

For coverage, is 100% statement coverage expected? There are a few that aren't tested, but they are only one lines returns like `return True` or `return "", ""`

**Comment by chrisjsewell** [comment]
@hukkin do you want to approve before I merge ta

**Pull request merged**